src/axom/quest/examples/quest_proe_bbox.cpp

```diff
@@ -0,0 +1,140 @@
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+
+#include "axom/CLI11.hpp"
+#include "axom/fmt.hpp"
+
+// _read_proe_include2_start
+#include "axom/mint/mesh/Mesh.hpp"
+#include "axom/mint/mesh/UnstructuredMesh.hpp"
+#include "axom/mint/utils/vtk_utils.hpp"  // for write_vtk
+// _read_proe_include2_end
+
+// _read_proe_include1_start
+#include "axom/quest/readers/ProEReader.hpp"
+// _read_proe_include1_end
+
+// _read_proe_typealiases_start
+namespace mint = axom::mint;
+namespace primal = axom::primal;
+namespace slic = axom::slic;
+
+using IndexType = axom::IndexType;
+using UMesh = mint::UnstructuredMesh<mint::SINGLE_SHAPE>;
+// _read_proe_typealiases_end
+
+//------------------------------------------------------------------------------
+void initialize_logger()
+{
+  // initialize logger
+  slic::initialize();
+  slic::setLoggingMsgLevel(slic::message::Info);
+
+  // setup the logstreams
+  std::string fmt = "";
+  slic::LogStream* logStream = nullptr;
+
+  fmt = "[<LEVEL>]: <MESSAGE>\n";
+  logStream = new slic::GenericOutputStream(&std::cout, fmt);
+
+  // register stream objects with the logger
+  slic::addStreamToAllMsgLevels(logStream);
+}
+
+//------------------------------------------------------------------------------
+void finalize_logger()
+{
+  slic::flushStreams();
+  slic::finalize();
+}
+
+struct Arguments
+{
+  std::string file_name;
+  std::string outfile_name;
+
+  void parse(int argc, char** argv, axom::CLI::App& app)
+  {
+    app
+      .add_option("-f,--file", this->file_name, "specifies the input mesh file")
+      ->check(axom::CLI::ExistingFile)
+      ->required();
+
+    app
+      .add_option("-o,--outfile",
+                  this->outfile_name,
+                  "specifies the output mesh file")
+      ->required();
+
+    app.get_formatter()->column_width(40);
+
+    // could throw an exception
+    app.parse(argc, argv);
+
+    slic::flushStreams();
+  }
+};
+
+int main(int argc, char** argv)
+{
+  initialize_logger();
+  Arguments args;
+  axom::CLI::App app {"Example showing how to read a Pro/E mesh"};
+
+  try
+  {
+    args.parse(argc, argv, app);
+  }
+  catch(const axom::CLI::ParseError& e)
+  {
+    int retval = -1;
+    retval = app.exit(e);
+    finalize_logger();
+    return retval;
+  }
+
+  SLIC_INFO("Reading file: '" << args.file_name << "'...\n");
+  // _read_proe_file_start
+  // Read file
+  axom::quest::ProEReader reader;
+  reader.setFileName(args.file_name);
+
+  // Set up a bounding box to keep only certain tets.
+  axom::quest::ProEReader::BBox3D bbox;
+  bbox.addPoint(axom::quest::ProEReader::Point3D {-1.5, -0.5, -0.5});
+  bbox.addPoint(axom::quest::ProEReader::Point3D {0, 1.5, 1.5});
+  // Keep only tets with all four nodes inside the bounding box.
+  reader.setTetPredFromBoundingBox(bbox, false);
+  // Pass true as the second argument of setTetPredFromBoundingBox() to
+  // keep tets with at least one node inside the bounding box.
+  // To keep all tets, do not set a TetPred.
```

Comment:
Suggestion: This example might be more useful if the user could provide the bounding box from the command line, e.g. using parameters named `--min` and `--max` that load 3 numbers each.

Reply:
done.